### PR TITLE
[iOS][Android][ads-admob][firebase-analytics][firebase-core][face-detector][google-sign-in] Upgrade native dependencies

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

This PR grew in size, because `iOS` dependencies coming from every module are combined in a single `Podfile` and therefore there's no way to do the upgrade incrementally (module by module). Incremental upgrades would cause build/CI failures until last affected module is taken care of.

Android part is splittable into multiple PRs, but to keep the changes aligned between platforms I would leave it here 👀 

Supersedes https://github.com/expo/expo/pull/16534
Resolves [ENG-4270](https://linear.app/expo/issue/ENG-4270/upgrade-native-firebase-dependencies)

# How

On `iOS`:
- Upgrading `Firebase/Core` triggered all other upgrades
- Modules affected:
    - `expo-firebase-core` (breaking changes are already resolved earlier)
        - `Firebase/Core@7.7.0 ➡️ 8.14.0`
    - `expo-face-detector` (no noticeable changes introduced): 
        - `GoogleMLKit/FaceDetection@2.1.0 ➡️ 2.6.0`:
        - `MLKitFaceDetection@1.2.0 ➡️ 1.5.0`
        - `MLKitCommon@2.1.0 ➡️ 5.0.0`
        - `MLKitVision@1.2.0 ➡️ 3.0.0`
    - `expo-ads-admob`
        - `Google-Mobile-Ads-SDK@7.69.0 ➡️ 8.13.0`:
            - `AdInterstitial` and `AdRewarded` migrated tot he new API
            - `AdInterstitial`'s `interstitialWillLeaveApplication`
            - `AdInterstitial` obtained new event `interstitialDidFailToOpen`
    - `expo-firebase-analytics`:
        - `Firebase/Core@7.7.0 ➡️ 8.14.0`
        - `setCurrentScreen` method is removed according to the [announcement from our docs](https://docs.expo.dev/versions/v44.0.0/sdk/firebase-analytics/#analyticssetcurrentscreenscreenname-screenclassoverride)
- I back-ported the changes to SDK 43 and SDK 44:
    - removed `expo-firebase-analytics#setCurrentScreen` method redirects the calls to the `logEvent` method with default `screen name` and `screen class` arguments.

On `Android`:
- Most work here is to align versions of native libraries to the iOS platform
- Modules affected:
    - `expo-face-detector`:
        - `androidx.exifinterface:exifinterface:1.0.0 ➡️ 1.3.3`
        - `com.google.firebase:firebase-ml-vision:24.0.1` and `com.google.firebase:firebase-ml-vision-face-model:19.0.0` ➡️ `com.google.mlkit:face-detection:16.1.5`
        - adjusted the codeabse to the new API
    - `expo-firebase-analytics`:
        - `com.google.firebase:firebase-core:17.2.1 ➡️ 20.1.2`
        - `com.google.firebase:firebase-common:19.0.0 ➡️ 20.1.0`
        - `com.google.firebase:firebase-analytics:17.2.1 ➡️ 20.1.2`
            - changes with `Firebase Instance ID SDK`: https://firebase.google.com/support/release-notes/android#analytics_v17-4-4
            - `setCurrentScreen` method removal
    - `expo-firebase-core`:
        - `com.google.firebase:firebase-core:17.2.1 ➡️ 20.1.2`
        - `com.google.firebase:firebase-common:19.0.0 ➡️ 20.1.0`
    - `expo-google-sign-in`:
        - added `com.google.android.gms:play-services-fitness` to read symbols that are no longer available via some transitive dependency chain
        - fixed the codebase and added removed constants directly to the code
    - `expo-ads-admob`:
        - `com.google.android.gms:play-services-ads:19.4.0 ➡️ 20.5.0`

### References:

- [`firebase` releases](https://firebase.google.com/support/releases)
- [`firebase` release notes for `iOS`](https://firebase.google.com/support/release-notes/ios)
- [`MLKit` release notes](https://developers.google.com/ml-kit/release-notes)
- [`Google Ad Mob iOS` release notes](https://developers.google.com/admob/ios/rel-notes)
- [`firebase` release notes for `Android`](https://firebase.google.com/support/release-notes/android)
- [`Google Ad Mob Android` release notes](https://developers.google.com/admob/android/rel-notes)

# Test Plan

- `iOS`:
    - [x] `bare-expo` and `Expo Go (versioned)` compile and run
    - [x] opening apps in `Expo Go` works  
    - [ ] ~`face-detection` works~ QA process 
    - [ ] ~`firebase-analytics` works~ QA process
    - [ ] ~`ads-admob` works~ QA process
- `Android`:
    - [x] `bare-expo` and `Expo Go (versioned)` compile and run
    - [x] opening apps in `Expo Go` works 
    - [ ] ~`face-detection` works~ QA process 
    - [ ] ~`firebase-analytics` works~ QA process
    - [ ] ~`ads-admob` works~ QA process

# TODOs

- [x] cleanup JS API according to native changes
- [x] cleanup docs
- [x] provide meaningful CHANGELOG entries

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobInterstitialAdModule.java [hunk 1]
```diff
@@ -72,124 +76,92 @@ public void setAdUnitID(String adUnitID, final Promise promise) {
...
-        if (mInterstitialAd != null && mInterstitialAd.isLoaded()) {
-          mShowAdPromise = promise;
-          mInterstitialAd.show();
-        } else {
-          promise.reject("E_AD_NOT_READY", "Ad is not ready", null);
-        }
+    new Handler(Looper.getMainLooper()).post(() -> {
+      if (mInterstitialAd != null) {
+        mShowAdPromise = promise;
+        mInterstitialAd.show(mActivityProvider.getCurrentActivity());
```
Comment (tsapeta) [inline packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobInterstitialAdModule.java:144]: Probably we should make sure that the current activity isn't `null`, the result is nullable

File: packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobRewardedVideoAdModule.java [hunk 2]
```diff
@@ -55,7 +46,8 @@ public enum Event {
     }
 
     @Override
-    public @NonNull String toString() {
+    public @NonNull
+    String toString() {
```
Comment (tsapeta) [inline packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobRewardedVideoAdModule.java:50]: This order looks weird 😅 
```suggestion
    @NonNull
    public String toString() {
```

File: packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobRewardedVideoAdModule.java [hunk 3]
```diff
@@ -91,99 +83,82 @@ public void setAdUnitID(String adUnitID, final Promise promise) {
...
-          @Override
-          public void onRewardedAdFailedToLoad(LoadAdError loadAdError) {
-            sendEvent(Event.DID_FAIL_TO_LOAD, AdMobUtils.createEventForAdFailedToLoad(loadAdError));
-            mRewardedAd = null;
-            mRequestAdPromise.reject("E_AD_REQUEST_FAILED", loadAdError.getMessage());
-            mRequestAdPromise = null;
-          }
-        });
-      }
+      RewardedAd.load(mActivityProvider.getCurrentActivity(), mAdUnitID, adRequest, new RewardedAdLoadCallback() {
```
Comment (tsapeta) [inline packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobRewardedVideoAdModule.java:98]: Same here, the activity can be `null`

**Review by tsapeta — APPROVED** [review]

File: packages/expo-face-detector/android/src/main/java/expo/modules/facedetector/FaceDetectorUtils.kt [hunk 4]
```diff
@@ -1,27 +1,34 @@
...
-      if (face.smilingProbability >= 0) {
-        putDouble("smilingProbability", face.smilingProbability.toDouble())
+      face.smilingProbability?.let {
+        if (it >= 0) {
+          putDouble("smilingProbability", it.toDouble())
+        }
       }
-      if (face.leftEyeOpenProbability >= 0) {
-        putDouble("leftEyeOpenProbability", face.leftEyeOpenProbability.toDouble())
+      face.smilingProbability?.let {
```
Comment (Kudo) [inline packages/expo-face-detector/android/src/main/java/expo/modules/facedetector/FaceDetectorUtils.kt:22]:
```suggestion
      face.leftEyeOpenProbability?.let {
```
i'm not pretty sure but it looks like a little strange here.

File: android/versioned-abis/expoview-abi44_0_0/src/main/java/abi44_0_0/expo/modules/facedetector/FaceDetectorUtils.kt [hunk 5]
```diff
@@ -1,27 +1,34 @@
...
-      if (face.smilingProbability >= 0) {
-        putDouble("smilingProbability", face.smilingProbability.toDouble())
+      face.smilingProbability?.let {
+        if (it >= 0) {
+          putDouble("smilingProbability", it.toDouble())
+        }
       }
-      if (face.leftEyeOpenProbability >= 0) {
-        putDouble("leftEyeOpenProbability", face.leftEyeOpenProbability.toDouble())
+      face.smilingProbability?.let {
```
Comment (Kudo) [inline android/versioned-abis/expoview-abi44_0_0/src/main/java/abi44_0_0/expo/modules/facedetector/FaceDetectorUtils.kt:22]:
```suggestion
      face.leftEyeOpenProbability?.let {
```

**Review by Kudo — APPROVED** [review]
not pretty sure about the ads request refactoring in detail. but mostly looks good to me.

File: packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobInterstitialAdModule.java (hunk 1 above)
Comment (bbarthec) [reply packages/expo-ads-admob/android/src/main/java/expo/modules/ads/admob/AdMobInterstitialAdModule.java:144]: Fair point, but as this is the last SDK we're supporting this module I would just leave it as is.
I guess there're lot more issues like this with module 👀

**Pull request merged**